src/schema/server_settings.py

```diff
@@ -99,87 +99,287 @@ class Difficulty(str, Enum):
     """
 
+    @classmethod
+    def setting_schema(cls, name: str) -> Dict[str, any]:
+        field = cls.__fields__.get(name)
+        if not field:
+            raise AttributeError(f"No field named {name}")
+        meta_setting = field.field_info.extra.get("meta_setting")
+        if not meta_setting:
+            raise AttributeError(f"Field {name} has no setting schema")
+        # See TODO on name
+        # meta_setting["default"] = field.default
+        meta_setting["serverSetting"] = True
+        return meta_setting
+
     # Language Generation Settings - Function calling
     default_function_capable_llm_model: str = Field("gpt-3.5-turbo", description="")
     default_function_capable_llm_temperature: float = Field(0.0, description="")
     default_function_capable_llm_max_tokens: int = Field(512, description="")
 
     # Language Generation Settings - Story telling
-    default_story_model: str = Field("gpt-3.5-turbo", description="")
-    default_story_temperature: float = Field(0.7, description="")
-    default_story_max_tokens: int = Field(512, description="")
+    default_story_model: str = Field(
+        "gpt-3.5-turbo",
+        meta_setting={
+            # Validated
+            "name": "default_story_model",
+            "label": "Story LLM Model",
+            "description": "Model used to generate story text.",
+            "type": "select",
+            "default": "gpt-3.5-turbo",
+            "options": [
+                {
+                    "value": "gpt-3.5-turbo",
+                    "label": "GPT 3.5 Turbo",
+                },
+                {
+                    "value": "gpt-4",
+                    "label": "GPT 4",
+                },
+            ],
+        }
+    )
+    default_story_temperature: float = Field(
+        0.7,
+        meta_setting={
+            # NEEDS WORK:
+            # TODO: Add a post-processing step to coerce this to a float.
+            "name": "default_story_temperature",
+            "label": "Story LLM Temperature",
+            "description": "Temperature (creativity-factor) for the narrative generation. 0=Robot, 1=Bonkers, 0.4=Default",
+            "type": "float",
+            "default": 0.7,
+        }
+    )
+    default_story_max_tokens: int = Field(
+        512,
+        meta_setting={
+            # NEEDS WORK:
+            # TODO: Add a post-processing step to coerce this to an int.
+            "name": "default_story_max_tokens",
+            "label": "Story LLM Max Tokens",
+            "description": "Maximum number of tokens permitted during generation. 256=Default",
+            "type": "int",
+            "default": 512,
+        }
+    )
 
     # Narration Generation Settings
     default_narration_model: str = Field("elevenlabs", description="")
 
     # Name
     name: Optional[str] = Field(
+        # TODO default is duplicated here, because I haven't thought through the scenario of what it defaults to on the
```

Comment:
Would it be possible to do something like the following?

- If python lets us describe something, use Python
- If Pydantic lets us describe something, use Pydantic
- As a last resort, use `meta_settings`

When we serialized out the UI configuration, that would just require us to something like:

```
output = {}
output.extend(meta_settings)
output.extend(pydantic_derived_settings)
output.extend(python_derived_settings)
return output
```

Reply:
What I was trying to say with this comment was that before these couple of PRs, the default value on ServerSettings differed from what the `ai-adventure-web` project had in the editor settings.  I *think* it should be fine to thunk down to what the `ai-adventure-web` project had, but was just leaving this to noodle on for a moment before making that call.

I agree though, what this would look like is to yank default out of `meta_settings` and put it in the pydantic `Field`.  Not sure if we even need the default concept on the web side if it's always reading from an agent?